lib/lroPoller.ts

```diff
@@ -105,15 +105,24 @@ export class LROPoller {
   /**
    * Send poll requests that check the LRO's status until it is determined that the LRO is finished.
    */
-  public async pollUntilFinished(): Promise<HttpOperationResponse> {
-    let result: Promise<HttpOperationResponse>;
+  public async pollUntilFinished(): Promise<RestResponse> {
+    let result: Promise<RestResponse>;
     const lroPollStrategy: LROPollStrategy | undefined = this._lroPollStrategy;
+
+    const initialResponse = this._initialResponse;
+    const operationSpec = initialResponse.request.operationSpec;
     if (!lroPollStrategy) {
-      result = Promise.resolve(this._initialResponse);
+      const getter = initialResponse.request.operationResponseGetter;
+      result = Promise.resolve(flattenResponse(initialResponse, getter && operationSpec && getter(operationSpec, initialResponse)));
     } else {
       result = lroPollStrategy.pollUntilFinished().then((succeeded: boolean) => {
         if (succeeded) {
-          return lroPollStrategy.getOperationResponse();
+          return lroPollStrategy.getOperationResponse().then(res => {
+            const response = this.getMostRecentResponse();
+            const getter = response.request.operationResponseGetter;
+            const responseSpec = getter && operationSpec && getter(operationSpec, response);
+            return flattenResponse(res, responseSpec);
```

Comment:
The previous three lines are functionally equivalent to lines 115 and 116. Can you extract them into a method that takes an `HttpOperationResponse` and returns a `Promise<RestResponse>`?

Reply:
On it

Reply:
Regarding line 121: Is there ever going to be a difference between using the `res` and accessing `getMostRecentResponse`? Shouldn't I just use `res` here?